### PR TITLE
Add 'sbp/domains/lock'

### LINKED ISSUE #2: Add a 'sbp/domains/lock' selector
### Problem

It's possible for unauthorized code to get access to the internal state of a domain that's bound to `this`.

### Solution

`'sbp/domains/lock'` which:

- Prevents further selectors from being registered on the supplied domain(s)
- When called with no arguments, locks all currently registered domains

### PR DESCRIPTION
**Pull request by snowteamer**
Closes #2

### CODE REVIEW

**Review by taoeffect — CHANGES_REQUESTED** [review]
While the implementation of `'sbp/domains/lock'` seems correct, this PR doesn't actually do any locking, but only gives the appearance of locking domains.

**Comment by snowteamer** [comment]
My bad, you're right. It seems like I forgot to update `'sbp/selectors/register'` accordingly.

File: index.js [hunk 1]
```diff
@@ -102,6 +109,19 @@ const SBP_BASE_SELECTORS = {
...
+  },
+  'sbp/domains/lock': function (domainNameOrNames: string | string[] | void) {
+    // If no argument was given then locks every known domain.
+    if (domainNameOrNames === undefined) {
+      for (const domain of Object.values(domains)) {
+        domain.locked = true
+      }
+    } else for (const name of typeof domainNameOrNames === 'string' ? [domainNameOrNames] : domainNameOrNames) {
+      if (!domains[name]) {
+        throw new Error(`SBP: unknown or invalid domain name: ${name}`)
```
Comment (taoeffect) [inline index.js:121]: Should probably simplify this error message to `SBP: cannot lock non-existent domain: ${name}`

**Review by taoeffect — APPROVED** [review]
Fantastic work @snowteamer!! Thanks for updating the tests too! 😄 

SBP is now one more step closer to perfection! 😁

File: index.js [hunk 2]
```diff
@@ -102,6 +109,19 @@ const SBP_BASE_SELECTORS = {
...
     if (!selectorFilters[selector]) selectorFilters[selector] = []
     selectorFilters[selector].push(filter)
+  },
+  'sbp/domains/lock': function (domainNameOrNames: string | string[] | void) {
+    // If no argument was given then locks every known domain.
+    if (domainNameOrNames === undefined) {
+      for (const domain of Object.values(domains)) {
+        domain.locked = true
+      }
+    } else for (const name of typeof domainNameOrNames === 'string' ? [domainNameOrNames] : domainNameOrNames) {
```
Comment (taoeffect) [inline index.js:119]: Hmm, this `else` should have a `{` after it, but I'll fix that in a followup.

File: dist/module.mjs [hunk 3]
```diff
@@ -41,8 +41,12 @@ const SBP_BASE_SELECTORS = {
   'sbp/selectors/register': function (sels) {
     const registered = []
     for (const selector in sels) {
-      const domain = domainFromSelector(selector)
-      if (selectors[selector]) {
+      const domainName = domainFromSelector(selector)
```
Comment (taoeffect) [inline dist/module.mjs:44]: This actually breaks line 59 above:

```js
if (selector === `${domain}/_init`) {
```

We both forgot to run `npm run flow` which caught this error.

**Pull request merged**